kratos/solving_strategies/strategies/arc_length_strategy.h

```diff
@@ -0,0 +1,689 @@
+        {
+            mDesiredIterations = rParameters["desired_iterations"].GetInt();
+            mMaxRadiusFactor   = rParameters["max_radius_factor"].GetDouble();
+            mMinRadiusFactor   = rParameters["min_radius_factor"].GetDouble();
+            mInitializeArcLengthWasPerformed = false;
+
+            // we initialize the list of load processes to be taken into account
+            if (rParameters["loads_sub_model_part_list"].size() > 0) {
+                mSubModelPartList.resize(rParameters["loads_sub_model_part_list"].size());
+                mVariableNames.resize(rParameters["loads_variable_list"].size());
+
+                if (mSubModelPartList.size() != mVariableNames.size())
+                    KRATOS_THROW_ERROR( std::logic_error, "For each SubModelPart there must be a corresponding nodal Variable", "")
+
+                for (unsigned int i = 0; i < mVariableNames.size(); i++) {
+                    mSubModelPartList[i] = &( model_part.GetSubModelPart(rParameters["loads_sub_model_part_list"][i].GetString()));
+                    mVariableNames[i] = rParameters["loads_variable_list"][i].GetString();
+                }
+            }
+        }
+
+    /**
+     * @brief Destructor.
+     * @details In trilinos third party library, the linear solver's preconditioner should be freed before the system matrix. We control the deallocation order with Clear().
+     */
+    ~ArcLengthStrategy() override = default;
+
+
+    /**
+     * @brief Initialization of member variables and prior operations
+     */
+    void Initialize() override
+    {
+        KRATOS_TRY;
+        BaseType::Initialize();
+        KRATOS_CATCH("");
+    }
+
+    /**
+     * @brief Clears the internal storage
+     */
+    void Clear() override
+    {
+        KRATOS_TRY;
+
+        SparseSpaceType::Clear(mpf);
+        SparseSpaceType::Clear(mpDxf);
+        SparseSpaceType::Clear(mpDxb);
+        SparseSpaceType::Clear(mpDxPred);
+        SparseSpaceType::Clear(mpDxStep);
+
+        TSystemVectorType& mf      = *mpf;
+        TSystemVectorType& mDxf    = *mpDxf;
+        TSystemVectorType& mDxb    = *mpDxb;
+        TSystemVectorType& mDxPred = *mpDxPred;
+        TSystemVectorType& mDxStep = *mpDxStep;
+
+        SparseSpaceType::Resize(mf, 0);
+        SparseSpaceType::Resize(mDxf, 0);
+        SparseSpaceType::Resize(mDxb, 0);
+        SparseSpaceType::Resize(mDxPred, 0);
+        SparseSpaceType::Resize(mDxStep, 0);
+
+        BaseType::Clear();
+
+        KRATOS_CATCH("");
+    }
+
+
+    /**
+     * @brief Performs all the required operations that should be done (for each step) before solving the solution step.
+     * @details A member variable should be used as a flag to make sure this function is called only once per step.
+     */
+    void InitializeSolutionStep() override
+    {
+        KRATOS_TRY;
+
+        if (!mInitializeArcLengthWasPerformed) {
```

Comment:
why don't you call the baseclass for this and in this fct only do things related to this class?

Reply:
I've removed it whenever I can. In this case it is necessary to differentiate between the base and the derived one (adhoc solution since we do not have the internal and external forces separated in different vectors...)

Reply:
hm ok

naturally I would call `BaseType::InitializeSolutionStep` first as it sets up the dofset etc but I am not familiar with the details of your impl